pages/api/auth/[...nextauth].ts

```diff
@@ -18,27 +18,43 @@ export const authOptions: AuthOptions = {
     }),
   ],
   callbacks: {
-    async jwt({ token, user, account }) {
-      // fetch user id from hasura
-      const client = new ApolloClient({
-        uri: "https://talented-duckling-40.hasura.app/v1/graphql/",
-        cache: new InMemoryCache(),
-      });
-      const { data } = await client.query({
-        query: FETCH_USER_BY_EMAIL,
-        variables: {
-          email: token.email,
-        },
-      });
+    async jwt({ token, user }) {
+
+      // FETCH user id from Hasura
+      if (user) {
+        const client = new ApolloClient({
+          uri: "https://talented-duckling-40.hasura.app/v1/graphql/",
+          cache: new InMemoryCache(),
+        });
+
+        try {
+          const { data } = await client.query({
+            query: FETCH_USER_BY_EMAIL,
+            variables: { email: user.email },
+          });
 
-      if (data.users?.length > 0) {
-        token.uid = data.users[0].id;
+          if (data?.users?.length > 0) {
+            
+            token.uid = data.users[0].id;
+            console.log("User found, UID set:", token.uid);
```

Comment:
don't merge in console logs, especially ones that print user data. it's fine to use when developing